src/validator.ts

```diff
@@ -1076,14 +1078,37 @@ function fnJoin(join: any, parts: any){
     return parts.join(join);
 }
 
-export function fnGetAtt(reference: string, attribute: string){
+export function fnGetAtt(reference: string, attributeName: string){
     if(workingInput['Resources'].hasOwnProperty(reference)){
         const resource = workingInput['Resources'][reference];
-        if (resource['Attributes'].hasOwnProperty(attribute)){
-            return resource['Attributes'][attribute];
-        } else if (resource['Type'].indexOf('Custom::') === 0 || resource['Type'] === 'AWS::CloudFormation::CustomResource') {
-            return `mockAttr_${reference}_${attribute}`;
-        } 
+        if (resource['Type'].indexOf('Custom::') === 0) {
+            return `mockAttr_${reference}_${attributeName}`;
+        } else if (resource['Type'] === 'AWS::CloudFormation::CustomResource') {
+            return `mockAttr_${reference}_${attributeName}`;
+        } else {
+            try {
+                // Lookup attribute
+                const attribute = resourcesSpec.getResourceTypeAttribute(resource['Type'], attributeName)
+                const primitiveAttribute = attribute as PrimitiveAttribute
+                if(!!primitiveAttribute['PrimitiveType']) {
```

Comment:
Is the `!!` intentional?

Reply:
Yup, that's my oldskool JS showing itself... 'double-not' coerces to boolean (i.e. true if not `falsy`). The intent here is to test for the existence of the property and make sure it has a value - I could have used `hasOwnProperty` (or instanceof) come to think of it, unless there is an idiomatic TypeScript way to do it...

Reply:
Great, thanks!